packages/terra-button-group/src/ButtonGroupButton.jsx

```diff
@@ -0,0 +1,59 @@
+import React, { PropTypes } from 'react';
```

Comment:
Yeah, my concern is with maintaining styles / react implementation of buttons in two places, `<Button />` and `<ButtonGroup.Button />`. I understand the UX concern of not wanting button link buttongroups and the concern of not wanting people to pass in props like href to `<ButtonGroup.Button />` and then having it render as an anchor.

I think we can avoid these with a h.o.c.  around `<Button />`. We can limit what variants are passed to it via the oneOfType propType check to ensure people can't use primary or link buttons. And for first class props button supports like `href` that we don't want to support in `<ButtonGroup.Button>`, we can intercept those props and set them to null so they don't get passed down to the wrapped `<Button />` component.